test/timer.ts

```diff
@@ -0,0 +1,145 @@
+import chai, { expect } from "chai";
+import proxyquire from "proxyquire";
+import sinon, { SinonSandbox } from "sinon";
+import sinonChai from "sinon-chai";
+import sinonTest from "sinon-test";
+import { DiscordInterface } from "../src/discord/interface";
+import logger from "../src/util/logger";
+import { DiscordWrapperMock } from "./mocks/discord";
+
+chai.use(sinonChai);
+const test = sinonTest(sinon);
+proxyquire.noPreserveCache();
+
+const discordMock = new DiscordWrapperMock();
+const discord = new DiscordInterface(discordMock, "mc!", logger);
+
+const countdownEntityStub = {
+	save: sinon.stub(),
+	remove: sinon.stub()
+};
+const Countdown = sinon.stub().returns(countdownEntityStub);
+const { PersistentTimer } = proxyquire("../src/timer", {
+	"./database/orm": { Countdown }
+});
+
+describe("PersistentTimer is a timer", function () {
+	beforeEach(function () {
+		sinon.resetHistory();
+	});
+
+	it(
+		"initializes correctly",
+		test(async function (this: SinonSandbox) {
+			const sendMessageSpy = this.spy(discord, "sendMessage"); // TODO: replace when we mock Discord the same way
+			const timer = await PersistentTimer.create(
+				discord,
+				new Date(Date.now() + 10000),
+				"1234567890",
+				"Test timer done",
+				1000
+			);
+			expect(timer.isActive()).to.be.true;
+			expect(sendMessageSpy).to.have.been.calledWith("1234567890", "Time left in the round: `00:10`");
+			expect(Countdown).to.have.been.called;
+			expect(countdownEntityStub.save).to.have.been.called;
+			expect(sendMessageSpy).to.have.been.calledBefore(countdownEntityStub.save);
+		})
+	);
+
+	it(
+		"calls tick and finishes",
+		test(async function (this: SinonSandbox) {
+			const sendMessageSpy = this.spy(discord, "sendMessage"); // TODO: replace when we mock Discord the same way
+			const edit = this.stub();
+			const getMessageStub = this.stub(discord, "getMessage").returns(
+				Promise.resolve({
+					id: "",
+					content: "",
+					attachments: [],
+					author: "",
+					channelId: "",
+					serverId: "",
+					reply: this.stub(),
+					react: this.stub(),
+					edit
+				})
+			);
+			const timer = await PersistentTimer.create(
```

Comment:
this is not a *big* deal and I understand the compulsion to save it to a variable, but github *is* yelling at me about the unused value warning here and a bit below.